compiler/src/Generate/JavaScript.hs

```diff
@@ -191,19 +192,35 @@ addGlobalHelp mode graph global state =
   let
     addDeps deps someState =
       Set.foldl' (addGlobal mode graph) someState deps
+    
+    argLookup = makeArgLookup graph
   in
   case graph ! global of
+    -- @LAMDERA
```

Comment:
Question not related to this PR: `@LAMDERA` annotations are there to help distinguish what was added in Lamdera but some of the annotations are for Lamdera features, while others (like this one) is for improvements to the Elm compiler. Would it be useful to have 2 annotations, one for Lamdera features and one for "improvements"? In order for these improvements to be easier to backport (to the Elm compiler, to elm-dev, ...)

Reply:
In this case it might be better we start a differences.md to track key modifications by PR or commit, rather than introduce varying annotations which still won’t give clear per-feature insight.

In this PR for example we’ve got a somewhat wide-sweeping and invasive that variable introduction, which is not the usual case for mods we’ve made so far. Hopefully that doesn't cause major pains when merging future upstream Elm changes.

But I can see Martin's tried to adhere to our code injection principles so seems he decided to draw the line somewhere re tagging/alternative implementation vs direct modification which is fair enough.

Reply:
@supermario I wonder whether what I did was a good tradeoff. Since that new argument needed to be threaded through everything, it seems like I'd have to make a new implementation of all those functions, mostly same but with the new arg, and dispatch to it in some parent call. That didn't seem worth it.